docs/features/software-templates/writing-templates.md

```diff
@@ -738,3 +740,118 @@ The `projectSlug` filter generates a project slug from a repository URL
 
 - **Input**: `github.com?repo=backstage&org=backstage`
 - **Output**: `backstage/backstage`
+
+## Custom Filters
+
+Whenever it is needed to extend the built-in filters with yours `${{ parameters.name | my-filter1 | my-filter2 | etc }}`, then you can add them
+using the property `addTemplateFilters` that you typically define using the `createRouter()` function of the `Scaffolder plugin`
+
+```ts title="packages/backend/src/plugins/scaffolder.ts"
+export default async function createPlugin({
+  logger,
+  config,
+}: PluginEnvironment): Promise<Router> {
+  ...
+  return await createRouter({
+    logger,
+    config,
+
+    additionalTemplateFilters: {
+        <YOUR_FILTERS>
+    }
+  });
+```
+
+The `addTemplateFilters` property accepts a `Record`
+
+```ts title="plugins/scaffolder-backend/src/service/Router.ts"
+  additionalTemplateFilters?: Record<string, TemplateFilter>;
+```
+
+where the first parameter is the name of the filter and the second `TemplateFilter` receives a list of `JSON value` arguments. The `templateFilter()` function must return a JsonValue (Json array, object or primitive).
+
+```ts title="plugins/scaffolder-node/src/types.ts"
+export type TemplateFilter = (...args: JsonValue[]) => JsonValue | undefined;
+```
+
+From a practical coding point of view, you will translate that into the following snippet code
+
+```ts title="packages/backend/src/plugins/scaffolder.ts"
+...
+  return await createRouter({
+    logger: env.logger,
+    config: env.config,
+    additionalTemplateFilters: {
+      base64: (...args: JsonValue[]) => btoa(args.join("")),
+      betterFilter: (...args: JsonValue[]) => { return `This is a much better string than "${args}", don't you think?` }
+    },
+});
+```
+
+And within your template, you will be able to use the filters like this
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: test
+  title: Test
+spec:
+  owner: user:guest
+  type: service
+
+  parameters:
+    - title: Test custom filters
+      properties:
+        userName:
+          title: Name of the user
+          type: string
+
+  steps:
+    - id: debug
+      name: debug
+      action: debug:log
+      input:
+        message: ${{ parameters.userName | betterFilter | base64 }}
+```
+
+### Register Custom Filters with the New Backend System
```

Comment:
I think that we should flip this around so that the `new backend system` is referred to as the default, and theres a `legacy backend system` section underneath too.

Reply:
Good idea. I will flip the 2 sections then

Reply:
What do you think about my refactoring ? [686e31b](https://github.com/backstage/backstage/pull/24451/commits/686e31b9614112cdd3ff20636464659de8de9511) @benjdlambert